### PR TITLE
Fix node events validation

### LINKED ISSUE #18460: Node events are broken
With https://github.com/kubernetes/kubernetes/pull/17819 all events targetted at nodes are rejected by the apiserver because https://github.com/kubernetes/kubernetes/blob/d42030170b2b20cb61cb79dff7d095998b7ee7b1/pkg/client/record/event.go#L289 assigns the `default` namespace to them:

```
E1209 14:58:49.481401      45 event.go:188] Server rejected event '&api.Event{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:api.ObjectMeta{Name:"192.168.65.40.141e4607d31461ef", GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil)}, InvolvedObject:api.ObjectReference{Kind:"Node", Namespace:"", Name:"192.168.65.40", UID:"192.168.65.40", APIVersion:"", ResourceVersion:"", FieldPath:""}, Reason:"RegisteredNode", Message:"Node 192.168.65.40 event: Registered Node 192.168.65.40 in NodeController", Source:api.EventSource{Component:"controllermanager", Host:""}, FirstTimestamp:unversioned.Time{Time:time.Time{sec:63585269929, nsec:479922159, loc:(*time.Location)(0x346c3e0)}}, LastTimestamp:unversioned.Time{Time:time.Time{sec:63585269929, nsec:479922159, loc:(*time.Location)(0x346c3e0)}}, Count:1, Type:"Normal"}': 'Event "192.168.65.40.141e4607d31461ef" is invalid: involvedObject.namespace: invalid value '', Details: namespace is not required for node' (will not retry!)
```

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #17819

@gmarek @sttts

Fixes #18460, https://github.com/mesosphere/kubernetes-mesos/issues/681

### CODE REVIEW

**Comment by stts** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

keliong@naver.com 님께 보내신 메일 <[kubernetes] Fix node events validation (#18462)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

**Comment by wojtek-t** [comment]
Sorry - I meant @sttts

**Comment by stts** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

keliong@naver.com 님께 보내신 메일 <Re: [kubernetes] Fix node events validation (#18462)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

**Comment by sttts** [comment]
lgtm

**Comment by sttts** [comment]
@k8s-bot unit test this

**Comment by erictune** [comment]
Allow both in validation, fix the client to not set "default", and eventually stop accepting "default".

**Comment by wojtek-t** [comment]
@erictune - PTAL

**Comment by sttts** [comment]
Agree, that's even better.

**Comment by sttts** [comment]
@wojtek-t maybe also touch the other code which sets the default namespace?

**Comment by sttts** [comment]
https://github.com/kubernetes/kubernetes/blob/d42030170b2b20cb61cb79dff7d095998b7ee7b1/pkg/client/record/event.go#L289

**Comment by sttts** [comment]
Or is there the danger that it will break compatibility with old apiserver? If there no validation out in the wild probably not though.

**Comment by wojtek-t** [comment]
I'm not sure we should change it in this PR.

**Comment by sttts** [comment]
At least with your patch it's better than in the release which has no validation at all. As a minimum please add a TODO(...) in the code.

File: pkg/api/validation/events.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func ValidateEvent(event *api.Event) validation.ErrorList {
 	allErrs := validation.ErrorList{}
 	// There is no namespace required for node.
 	if event.InvolvedObject.Kind == "Node" &&
+		event.Namespace != api.NamespaceDefault &&
```
Comment (erictune) [inline pkg/api/validation/events.go]: // Older client code accidentally set event.Namespace to api.NamespaceDefault, so we accept that too, but "" is preferred.

**Comment by erictune** [comment]
lgtm.

**Comment by erictune** [comment]
@lavalamp FYI.

**Comment by sttts** [comment]
@erictune thanks for ignoring my comment. Two acceptable namespaces and a not matching error message deserves a code comment.

**Comment by sttts** [comment]
@erictune sorry. iOS github client didn't show your comment. would make sense to have it in the code though.

File: pkg/api/validation/events.go (hunk 1 above)
Comment (mqliang) [inline pkg/api/validation/events.go]: +1 for adding a comment

**Comment by wojtek-t** [comment]
@erictune - PTAL

**Comment by erictune** [comment]
lgtm

**Comment by gmarek** [comment]
@k8s-bot test this

**Pull request merged**